syntax = "proto3";

package imageclassifier;

service ImageStreamer {
  rpc StreamImages(Empty) returns (stream ImageRequest);
}

service ImageClassifier {
  rpc Classify(stream ImageRequest) returns (stream ClassificationResult);
}

message Empty {}

message ImageRequest {
  string image_id  = 1;      
  bytes  image_data = 2;     
}

message ClassificationResult {
  string image_id  = 1;
  string label     = 2;
  float  confidence = 3;
}